AndelaOSP/wire-api
# #164040132 Set up Websocket Communication

#### What does this PR do?

This PR adds Websocket connection support which is needed for subsequent notification stories. It also add new models to the database. This PR is worked on in tandem with this [frontend PR](https://github.com/AndelaOSP/wire/pull/179) so they share similar information.

#### Description of Task to be completed?

- Changes server implementation to accommodate Websocket connection
- Added support for authenticating socket clients
- Added a proxy for making websocket work with webpack-dev-server
- Created new tables and join table for notification stories
- Added a simple websocket example of the notification

#### How should this be manually tested?

- Migrate the new models with `sequelize db:migrate`
- Navigate to an incident's timeline page.
- Select your name on the CC list.
- You should see the notification button increment.

#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories?

[#164040132](https://www.pivotaltracker.com/story/show/164040132)

#### Screenshots (if appropriate)

N/A
